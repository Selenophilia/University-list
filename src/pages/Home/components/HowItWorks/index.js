import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import { Card, CardContent, Typography } from "@material-ui/core";
import WebIcon from "@material-ui/icons/Web";
import AssignmentIndIcon from "@material-ui/icons/AssignmentInd";
import EmailIcon from "@material-ui/icons/Email";

const useStyles = makeStyles((theme) => ({
  howitworks: {
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    flexDirection: "column",
    marginBottom: 50
  },
  title: {
    marginBottom: 30
  },
  line: {
    color: "#000",
    background: "#000",
    border: "2px solid #000",
    width: 150,
    opacity: 0.8
  },
  cardTitle: {
    fontSize: 34,
    fontWeight: "bold"
  },
  cardText: {
    fontSize: 20
  },
  text: {
    fontSize: 32,
    fontWeight: "bold",
    marginBottom: 10
  },
  icon: {
    fontSize: 54
  },
  cards: {
    display: "flex ",
    justifyContent: "space-around",
    width: "100%",
    // paddingLeft: 50,
    // paddingRight: 50,
    [theme.breakpoints.down("md")]: {
      alignItems: "center",
      flexDirection: "column"
    }
  },
  card: {
    padding: 20,
    width: 400,
    height: 350,
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    flexDirection: "column",
    textAlign: "justify",
    borderRadius: 8,
    [theme.breakpoints.down("md")]: {
      marginBottom: 20
    }
  }
}));

const HowItWorks = () => {
  const classes = useStyles();
  return (
    <div className={classes.howitworks}>
      <div className={classes.title}>
        <Typography variant="h2" className={classes.text}>
          How it works
        </Typography>
        <hr className={classes.line} />
      </div>
      <div className={classes.cards}>
        <Card className={classes.card} variant="outlined">
          <CardContent className={classes.card}>
            <WebIcon className={classes.icon} />
            <Typography variant="h5" className={classes.cardTitle}>
              Search
            </Typography>
            <Typography variant="h6" className={classes.cardText}>
              Browse across our website to find what best universities you like
              with ease.
            </Typography>
          </CardContent>
        </Card>
        <Card className={classes.card} variant="outlined">
          <CardContent className={classes.card}>
            <EmailIcon className={classes.icon} />
            <Typography variant="h5" className={classes.cardTitle}>
              Apply
            </Typography>
            <Typography variant="h6" className={classes.cardText}>
              After searching pick a university you likea and apply by going to
              their website!
            </Typography>
          </CardContent>
        </Card>
        <Card className={classes.card} variant="outlined">
          <CardContent className={classes.card}>
            <AssignmentIndIcon className={classes.icon} />
            <Typography variant="h5" className={classes.cardTitle}>
              Enroll
            </Typography>
            <Typography variant="h6" className={classes.cardText}>
              Lastly, after submitting your application enroll!
            </Typography>
          </CardContent>
        </Card>
      </div>
    </div>
  );
};

export default HowItWorks;
