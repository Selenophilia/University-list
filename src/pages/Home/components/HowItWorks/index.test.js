import React from "react";
import { render } from "@testing-library/react";
import HowItWorks from ".";

const renderRoot = () => {
  return render(<HowItWorks />);
};

it("renders HowItWorks component correctly ", () => {
  const { asFragment } = renderRoot();
  expect(asFragment()).toMatchSnapshot();
});
