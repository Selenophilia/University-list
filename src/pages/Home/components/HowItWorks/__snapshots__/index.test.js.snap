// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders HowItWorks component correctly  1`] = `
<DocumentFragment>
  <div
    class="makeStyles-howitworks-1"
  >
    <div
      class="makeStyles-title-2"
    >
      <h2
        class="MuiTypography-root makeStyles-text-6 MuiTypography-h2"
      >
        How it works
      </h2>
      <hr
        class="makeStyles-line-3"
      />
    </div>
    <div
      class="makeStyles-cards-8"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-card-9 MuiPaper-outlined MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root makeStyles-card-9"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-icon-7"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm-5 14H4v-4h11v4zm0-5H4V9h11v4zm5 5h-4V9h4v9z"
            />
          </svg>
          <h5
            class="MuiTypography-root makeStyles-cardTitle-4 MuiTypography-h5"
          >
            Search
          </h5>
          <h6
            class="MuiTypography-root makeStyles-cardText-5 MuiTypography-h6"
          >
            Browse across our website to find what best universities you like with ease.
          </h6>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiCard-root makeStyles-card-9 MuiPaper-outlined MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root makeStyles-card-9"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-icon-7"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
            />
          </svg>
          <h5
            class="MuiTypography-root makeStyles-cardTitle-4 MuiTypography-h5"
          >
            Apply
          </h5>
          <h6
            class="MuiTypography-root makeStyles-cardText-5 MuiTypography-h6"
          >
            After searching pick a university you likea and apply by going to their website!
          </h6>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiCard-root makeStyles-card-9 MuiPaper-outlined MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root makeStyles-card-9"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-icon-7"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3h-4.18C14.4 1.84 13.3 1 12 1c-1.3 0-2.4.84-2.82 2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-7 0c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm0 4c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm6 12H6v-1.4c0-2 4-3.1 6-3.1s6 1.1 6 3.1V19z"
            />
          </svg>
          <h5
            class="MuiTypography-root makeStyles-cardTitle-4 MuiTypography-h5"
          >
            Enroll
          </h5>
          <h6
            class="MuiTypography-root makeStyles-cardText-5 MuiTypography-h6"
          >
            Lastly, after submitting your application enroll!
          </h6>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
